docs/roadmap.md

```diff
@@ -0,0 +1,122 @@
+## Planned Links
+
+### Terminating:
+
+* Websocket: subscriptions support
+* Batching-\*: support for batching operations with a transport
+
+### Intermediate:
+
+* Caching: returns data if result in the cache and stores data in cache on response
+* Catch: catch all errors and modify the Observable to return something else
+* Dedup: saves query signatures that are currently on the wire and returns the result for all of those queries
+* Batch: batches operations then performs a function on the batched operations
+
+### Adapter (not a Link):
+
+* Backwards-compatibility Wrapper: exposes `query`, `mutate`, and `subscribe`
+
+## Use Cases Informing Future Development
+
+Apollo Links are designed to be used as an isomorphic stand-alone extensible GraphQL client.
+In addition, links fit all current and anticipated needs of Apollo Client.
+As such, links are targeted towards the following use-cases:
+
+* Do a simple GraphQL query with it that resolves only once
+* Do a simple GraphQL mutation that resolves only once
+* Do a subscription
+* Do a GraphQL query that contains @defer, @stream or @live
+  * Support custom annotations, such as @connection
+
+<br>
+
+* Modify query, variables, operationName and context
+  * Use persistent queries
+* Include extra metadata to server along with queries
+  * Include component sending query
+  * Send time stamp of query leaving transport
+* Receive extra metadata with the query result
+  * Send caching information
+
+<br>
+
+* Use HTTP transport, websocket transport, hybrid transport
+* Use alternative transports
+  * File System
+  * DDP
+* Mock partial or full response
+* Route operations to alternate endpoints
+  * Partial operations to REST endpoint
+  * Partial operations to GraphQL endpoint
+  * Partial operations to Mock
+* Split single operation into multiple operations
+  * Race one operation to local storage and another to network
+* Per request retries based on network status or response
+* Polling
+* Batching at the transport/fetch layer
+  * Perform batch operations within a composed link
+* Deduplicate of on-the-wire requests
+* Re-route and modify errors
+  * transform network error into GraphQL error
+
+<br>
+
+* Network layer caching
```

Comment:
Yeah! Thank you for adding to the discussion. Allowing consumers to pick functionality is certainly a goal. Inverting the relationship is a possible future option, meaning apollo-client could become the synchronous adapter on top of a bunch of `apollo-link`s. Any further thoughts would be really appreciated and informative for future development